### ISSUE TITLE
GL_Out_Of_Memory when alternating between two image hrefs on ImageNode

### ISSUE THREAD

**Issue by Uli-K** [libavg/libavg#762]
When I switch repeatetly between two images, assigning two file paths to an avg.ImageNode's href I get GL_OUT_OF_MEMORY

Here is the code that triggers the problem:
```
from libavg import app, avg, player


class ImageTest(app.MainDiv):

    def onInit(self):
        self.__image = avg.ImageNode(
            parent=self,
            pos=(50, 50),
            size=(200, 100),
            href="sample1.png"
        )

        player.subscribe(player.KEY_UP, self.__on_key_up)

    def __on_key_up(self, event):
        if event.keyname == "I":
            for i in range(0, 101):
                if self.__image.href == "sample1.png":
                    self.__image.href = "sample2.png"
                else:
                    self.__image.href = "sample1.png"

if __name__ == '__main__':
    app.App().run(ImageTest(), app_resolution="800,600")
```

on my mashine it crashes after running it the second time around (i.e., pressing "I" twice)
PC specification: Windows, 64bit, Intel i7, 16GB RAM with NVIDIA GeForce GTX 670

```
Error Message:
[16-03-21 18:19:10.172][.ERR][.....NONE] : OpenGL error in End of frame: GL_OUT_OF_MEMORY
Traceback (most recent call last):
  File ".../image-test.py", line 27, in <module>
    app.App().run(ImageTest(), app_resolution="800,600")
  File "C:\Python27\lib\site-packages\libavg\app\app.py", line 119, in run
    self._runLoop()
  File "C:\Python27\lib\site-packages\libavg\app\app.py", line 397, in _runLoop
    libavg.player.play()
libavg.avg.Exception: Assertion failed in ..\..\src\graphics\GLContext.cpp: 476


Process finished with exit code 1
```

Thank you :)

**Comment by uzadow** [comment]
I can reproduce this here (only on windows), but it's a bit harder to fix. What is happening is that libavg is queueing up lots of texture load/unload commands in GLContextManager, and when it comes around to executing them, most of the commands don't make sense anymore. 

As a workaround: Increase the GPU cache size in avgrc.

To fix: GLContextManager needs to index all pending commands by the GPU resource used. When a new command comes in, old commands referencing the same resource become obsolete and are deleted.

**Comment by uzadow** [comment]
Fixed with e0cb935f27.
